presto-product-tests/conf/docker/common/compose-commons.sh

```diff
@@ -3,7 +3,8 @@
 # so we canonicalize them.
 function canonical_path() {
     # make the path start with '/' or './'. Otherwise the result for 'foo.txt' is an absolute path.
-    local PATH=`[[ $1 = /* ]] && echo "$1" || echo "./${1#./}"`
+    local PATH_REFERENCE=$1
+    local PATH=`[[ ${!PATH_REFERENCE} = /* ]] && echo "${!PATH_REFERENCE}" || echo "./${!PATH_REFERENCE#./}"`
```

Comment:
Also add a comment about what the `${!` is doing, or simplify if possible. Even after reading the manual (and I'm fairly experienced with shell scripting), it's not clear what what this does. non-bash experts will have no chance.

```
${!name[@]}
${!name[*]}
       If name is an array variable,  expands  to  the  list  of  array
       indices  (keys)  assigned  in  name.   If  name is not an array,
       expands to 0 if name is set and null otherwise.  When @ is  used
       and the expansion appears within double quotes, each key expands
       to a separate word.